Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php

```diff
@@ -54,16 +53,11 @@ public function process(ContainerBuilder $container)
 
                     $parameters = $method->getParameters();
 
-                    // if no param or optional param, skip
-                    if (count($parameters) !== 1 || $parameters[0]->isOptional()) {
+                    // if no param, optional param or non-class param, skip
+                    if (count($parameters) !== 1 || $parameters[0]->isOptional() || $parameters[0]->getClass() === null) {
```

Comment:
Could it be useful to create a method named correctly  for this if so we can remove the comment, make the code more clean and readable? What do you think?

Reply:
Was thinking the same. But I'd rather extract all the logic the determines whether a given `ReflectionMethod` is a valid handler. For my taste the whole process method is too long and nested but I didn't want to dig it all over for this small change.

Reply:
Ok thanks for the explanation